### PR TITLE
feat: testing CI job

### PR DESCRIPTION
**Pull request by benceharomi**
# What ❔
* added a CI job that runs tests
* temporarily excluded failing tests

## Why ❔
To automatically run tests with every PR.

## Checklist
- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: test/KnownCodesStorage.spec.ts [hunk 1]
```diff
@@ -1,157 +1,157 @@
...
+//     BOOTLOADER_FORMAL_ADDRESS,
+//     EMPTY_STRING_KECCAK,
+//     L1_MESSENGER_SYSTEM_CONTRACT_ADDRESS,
+//     COMPRESSOR_CONTRACT_ADDRESS
+// } from './shared/constants';
+// import { Wallet } from 'zksync-web3';
+// import { getWallets, deployContract, loadArtifact, setCode, getCode } from './shared/utils';
+// import { network, ethers } from 'hardhat';
+
+// describe('KnownCodesStorage tests', function () {
```
Comment (mm-zk) [inline test/KnownCodesStorage.spec.ts:13]: can you try 'xdescribe'? ( it is supposed to turn off the test, and print info that it was skipped)

File: test/ImmutableSimulator.spec.ts [hunk 2]
```diff
@@ -33,27 +33,27 @@ describe('ImmutableSimulator tests', function () {
             );
         });
 
-        it('successfully set', async () => {
-            await network.provider.request({
-                method: 'hardhat_impersonateAccount',
-                params: [DEPLOYER_SYSTEM_CONTRACT_ADDRESS]
-            });
+        // it('successfully set', async () => {
```
Comment (mm-zk) [inline test/ImmutableSimulator.spec.ts:36]: and 'xit' here

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -1,17 +1,72 @@
...
+          node-version: 18.18.0
+          cache: yarn
+
+      - name: Use era-test-node for testing
+        uses: dutterbutter/era-test-node-action@latest
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Restore artifacts cache
```
Comment (mm-zk) [inline .github/workflows/ci.yaml:59]: oh - very nice.. this cache is per repo? or per this ci job?

**Review by mm-zk — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (benceharomi) [reply .github/workflows/ci.yaml:59]: It creates one per github action, just so we could use it in both of the build and test jobs without having to compile twice.

File: test/KnownCodesStorage.spec.ts (hunk 1 above)
Comment (benceharomi) [reply test/KnownCodesStorage.spec.ts]: Of course, good idea! Updated everything accordingly.

**Review by mm-zk — APPROVED** [review]

**Pull request merged**